quora-api/src/main/java/com/upgrad/quora/api/controller/UserController.java

```diff
@@ -40,12 +46,42 @@
         userEntity.setContactNumber(signupUserRequest.getContactNumber());
         userEntity.setSalt("1234abc");
 
-
-        /*final UserEntity createdUserEntity = userService.signup(userEntity);
+        final UserEntity createdUserEntity = userService.signup(userEntity);
         SignupUserResponse userResponse = new SignupUserResponse();
         userResponse.setId(createdUserEntity.getUuid());
         userResponse.setStatus("USER SUCCESSFULLY REGISTERED");
-        return new ResponseEntity<SignupUserResponse>(userResponse, HttpStatus.CREATED);*/
-        return null;
+        return new ResponseEntity<SignupUserResponse>(userResponse, HttpStatus.CREATED);
+    }
+
+    @RequestMapping(method = RequestMethod.POST, path = "user/signin", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<SigninResponse> signin(@RequestHeader("authorization") final String authorization) throws AuthenticationFailedException {
+        byte[] decode = Base64.getDecoder().decode(authorization.split("Basic ")[1]);
+        String decodedText = new String(decode);
+        String[] decodedArray = decodedText.split(":");
+
+        UserAuthTokenEntity userAuthToken = userService.signIn(decodedArray[0], decodedArray[1]);
```

Comment:
This class is renamed as UserAuthEntity. Please change this.

Reply:
Changed class name to UserAuthEntity